www/webapp/static/js/login_flow.js

```diff
@@ -0,0 +1,184 @@
+      signUpForm[i].classList.remove("d-none");
+      requiredSignUpInputs();
+    }
+    checkedEmail = true;
+    btnSubmit.innerText = "Sign Up";
+    mainTitle.innerText = "Please fill the sign up form";
+    initialEmail = emailInput.value;
+    return;
+  }
+  body = await response.json();
+  if (body.sso_provider) {
+    errorMessage.innerText = "You are already registered with " + body.sso_provider;
+    errorMessage.classList.remove("d-none");
+    initialEmail = emailInput.value;
+    checkedEmail = true;
+    existUser = true;
+    return;
+  }
+
+  for (let i = 0; i < signInForm.length; i++) {
+    signInForm[i].classList.remove("d-none");
+    requiredSignInInput();
+  }
+  checkedEmail = true;
+  existUser = true;
+  initialEmail = emailInput.value;
+  btnSubmit.innerText = "Sign In";
+  mainTitle.innerText = "Please enter your password";
+}
+
+async function loginForm() {
+  const email = document.getElementById("email").value;
+  const password = document.getElementById("password").value;
+
+  
+  const formData = {
+    email: email,
+    password: password,
+  };
+
+  const response = await fetch("/api/auth/sign_in", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(formData),
+  });
+
+  if (!response.ok) {
+    errorMessage.innerText = "Invalid email or password";
+    errorMessage.classList.remove("d-none");
+    return;
+  }
+
+  const data = await response.json();
+  const token = data.token;
+  if (token) {
+    sessionStorage.setItem("token", token);
+    window.location.href = "/";
+  }
+}
+
+async function registerForm() {
+  const given_name = document.getElementById("given_name").value;
+  const surname = document.getElementById("surname").value;
+  const email = document.getElementById("email").value;
+  const password = document.getElementById("password").value;
+
+  const formData = {
+    email: email,
+    password: password,
+    given_name: given_name,
+    surname: surname,
+  };
+
+  const response = await fetch("/api/auth/sign_up", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(formData),
+  });
+
+  if (!response.ok) {
+    errorMessage.innerText = "Already taken the user";
+    errorMessage.classList.remove("d-none");
+    return;
+  }
+  loginForm();
+}
+
+emailInput.addEventListener('input', function() {
+  const currentEmail = emailInput.value;
+
+  if (currentEmail !== initialEmail && checkedEmail) {
+    resetForm();
+  }
+});
+
+document.querySelector("form").addEventListener("submit", function (event) {
+  event.preventDefault(); // Prevent the default form submission
+  errorMessage.classList.add("d-none");
+  switch (true) {
+    case !checkedEmail:
+      checkMail();
+      break;
+    case existUser && checkedEmail:
+      loginForm();
+      break;
+    case !existUser && checkedEmail:
+      registerForm();
+      break;
+    default:
+      break;
+  }
+});
```

Comment:
Refactor this with an enum

```suggestion
function changeSubmitButtonAction(action) {
   const actionButton = document.getActionButton();
   
   actionButton.attr('button-action', action);
}
const SubmitActions = {
    CHECK_EMAIL: 0,
    LOGIN: 1,
    REGISTER: 2
}
submitButton.addEventListener("submit", function (event) {
   const actions = {
     SubmitActions.CHECK_EMAIL: checkEmail,
     SubmitActions.LOGIN: loginForm,
     SubmitActions.REGISTER: registerForm
   }
   const currentAction = submitButton.getAttribute("button-action")
   const actionFunc = actions[`${currentAction}`] || () => { diplay error action not found}
   
   return actuonFunc()
}
```